File: app/src/main/java/com/asmat/rolando/bakingapp/adapters/RecipeStepsViewPagerAdapter.java
package com.asmat.rolando.bakingapp.adapters;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.asmat.rolando.bakingapp.R;
import com.asmat.rolando.bakingapp.fragments.CompleteStepFragment;
import com.asmat.rolando.bakingapp.models.Recipe;

/**
 * Created by rolandoasmat on 8/13/17.
 */

public class RecipeStepsViewPagerAdapter extends FragmentStatePagerAdapter {
    private Recipe recipe;
    private Context mContext;

    @Override
    public int getCount() {
        return this.recipe.getSteps().length;
    }

    @Override
    public Fragment getItem(int position) {
        return CompleteStepFragment.Companion.newInstance(recipe, position);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        int id = this.recipe.getSteps()[position].getId();
        if(id == 0) {
            return this.mContext.getResources().getString(R.string.intro);
        } else {
            return String.valueOf(id);
        }
    }

    public void setRecipe(Recipe recipe) {
        this.recipe = recipe;
    }

    public RecipeStepsViewPagerAdapter(FragmentManager fm, Context mContext) {
        super(fm);
        this.mContext = mContext;
    }
}

File: app/src/main/java/com/asmat/rolando/bakingapp/adapters/RecipeStepsViewPagerAdapter.kt
package com.asmat.rolando.bakingapp.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v4.view.PagerAdapter
import com.asmat.rolando.bakingapp.fragments.CompleteStepFragment
import com.asmat.rolando.bakingapp.models.Recipe

/**
 * Created by rolandoasmat on 7/14/17.
 */
// https://github.com/google/ExoPlayer/issues/591
class RecipeStepsViewPagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {
    var recipe: Recipe? = null

    override fun getCount(): Int {
        val count = recipe?.steps?.size
        if(count != null) {
            return count
        } else {
            return 0
        }
    }

    override fun getItem(position: Int): Fragment {
        return CompleteStepFragment.newInstance(recipe!!, position)
    }

    override fun getPageTitle(position: Int): CharSequence {
        val id =  recipe!!.steps[position].id
        if(id == 0) {
            return "Intro"
        } else {
            return id.toString()
        }
    }
}